lib/zlib.js

```diff
@@ -418,7 +419,10 @@ function processChunkSync(self, chunk, flushFlag) {
   const chunkSize = self._chunkSize;
 
   let error;
-  self.on('error', function onError(er) {
+  self.on('error', () => {
+    // Block unhandled exception error.
+  });
```

Comment:
Why this change? I would try to avoid such a drastic measure.

Reply:
Because we need the error synchronously in the same tick, while stream emit it in next tick.

Reply:
I could make a different version of this that doesn't use `kSyncError` and instead uses a property on the object, if that preferable? Still needs an `'error'` handler here though to not cause unhandled exception.

Reply:
> Because we need the error synchronously in the same tick, while streams/destroy(err)/errorOrDestroy emit it in next tick.

Why cannot we have zlib emit on nextTick?

Reply:
`processChunkSync` is supposed to `throw` synchronously. I updated the PR with maybe a more preferable version of this.